src/fortran/psychrolib.f90

```diff
@@ -821,6 +823,57 @@ function GetDryAirVolume(TDryBulb, Pressure) result(DryAirVolume)
     end if
   end function GetDryAirVolume
 
+  function GetTDryBulbFromEnthalpyAndHumRatio(MoistAirEnthalpy, HumRatio) result(TDryBulb)
+    !+ Return dry bulb temperature from enthalpy and humidity ratio.
+    !+ Reference:
+    !+ ASHRAE Handbook - Fundamentals (2017) ch. 1 eqn 30
+    !+ Notes:
+    !+ Based on the `GetMoistAirEnthalpy` function, rearranged for humidity ratio.
+
+    real, intent(in)  ::  MoistAirEnthalpy
+      !+ Moist air enthalpy in Btu lb⁻¹ [IP] or J kg⁻¹
+    real, intent(in)  ::  HumRatio
+      !+ Humidity ratio in lb_H₂O lb_Air⁻¹ [IP] or kg_H₂O kg_Air⁻¹ [SI]
+    real              ::  TDryBulb
+      !+ Dry-bulb temperature in °F [IP] or °C [SI]
+    real              ::  BoundedHumRatio
+    !+ Local function humidity ratio bounded to no less than 1.0E-5 kgH₂O kgAIR⁻¹
+
+    if (HumRatio < 0.0) then
+      error stop "Error: humidity ratio is negative"
+    end if
+
+    ! Bound humidity ratio to no less than 1.0E-5 kgH₂O kgAIR⁻¹
+    BoundedHumRatio = max(HumRatio, 1.0E-5)
```

Comment:
If you use a bounded humidity ratio in Fortran, it should be used in all languages - it's not used in Python at the moment.